src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.MultiSegment.cs

```diff
@@ -2380,25 +2379,48 @@ private bool ConsumeCommentMultiSegment()
             }
 
             byte marker = localBuffer[0];
-
-            if (marker == JsonConstants.Slash)
+            if (marker != JsonConstants.Slash && marker != JsonConstants.Asterisk)
             {
-                return ConsumeSingleLineCommentMultiSegment(localBuffer.Slice(1), leftOver, start, _consumed);
+                ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.InvalidCharacterAtStartOfComment, marker);
             }
 
-            if (marker != JsonConstants.Asterisk)
+            localBuffer = localBuffer.Slice(1);
+            if (localBuffer.Length == 0)
             {
-                ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.ExpectedStartOfValueNotFound, marker);
+                if (IsLastSpan)
+                {
+                    ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.ExpectedComment);
+                }
+                if (!GetNextSpan())
+                {
+                    if (IsLastSpan)
+                    {
+                        ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.ExpectedComment);
```

Comment:
This is still inconsistent between the single-segment and multi-segment case. It's odd that we use `ExceptionResource.ExpectedComment` only in the multi-segment case and never in `ConsumeComment` for the single-segment case.

I think we should just use `ExceptionResource.ExpectedComment` in all cases and remove `UnexpectedEndOfDataWithinStartOfComment` (similar to what I mentioned previously). https://github.com/dotnet/corefx/pull/35934/files#r277941907

If you insist we keep both, at the very least, we need to be consistent between the single-segment and multi-segment case, in terms of what exception message is visible to the user.
https://github.com/dotnet/corefx/blob/8fcdc1a31464bf2627ab95736984d53014d53131/src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.cs#L2235
https://github.com/dotnet/corefx/blob/8fcdc1a31464bf2627ab95736984d53014d53131/src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.cs#L2241

https://github.com/dotnet/corefx/blob/8fcdc1a31464bf2627ab95736984d53014d53131/src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.MultiSegment.cs#L2392
https://github.com/dotnet/corefx/blob/8fcdc1a31464bf2627ab95736984d53014d53131/src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.MultiSegment.cs#L2398